server/service/individualServices/SoftwareUpgrade.js

```diff
@@ -27,10 +27,8 @@ var traceIncrementer = 1;
  * **/
 
 exports.upgradeSoftwareVersion = async function (user, xCorrelator, traceIndicator, customerJourney, _traceIncrementer) {
-    if (traceIndicator == 0) {
-        traceIndicator = 1;
-    }else{
-        traceIndicator =_traceIncrementer
+    if (_traceIncrementer !== 0) {
```

Comment:
As per the commit, https://github.com/openBackhaul/AdministratorAdministration/commit/9aabbbd03cdf2443a90be85df333f4bb92a30cb0, the code has been changed from the current change to existing code, could you please check for the reason for that change and provide a justification.

Reply:
as the per previous commit, i have not declare the variable traceIndicator has a global variable so i was receiving traceIndicator has 0 and i am assigning  to 1 . but now i made code optimization

Reply:
when _traceIncrementer !== 0, it means there is some value existing to the incoming traceIncrementor. We shall revisit the changes , something like if(_traceIncrementer == 0) or if(!_traceIncrementer) then to initiate _traceIncrementer = traceIncrementor. 

Also, kindly consider the calling function to decide the final snippet.